plugins/single_plugins/preserve-output.cpp

```diff
@@ -0,0 +1,314 @@
+            if (!view_has_data(view))
+            {
+                view_store_data(view, output, i);
+            }
+        }
+    };
+
+    wf::signal_connection_t output_removed = [=] (wf::signal_data_t *data)
+    {
+        auto signal_data = (wf::output_removed_signal*)data;
+        LOGD("Received output-removed event: ", signal_data->output->to_string());
+        outputs_being_removed.erase(std::remove(
+            outputs_being_removed.begin(),
+            outputs_being_removed.end(),
+            signal_data->output->to_string()), outputs_being_removed.end());
+        LOGD("Outputs currently being removed: ");
+        for (auto el : outputs_being_removed)
+        {
+            LOGD("- ", el);
+        }
+    };
+
+    wf::signal_connection_t restore_views_to_output = [=] (wf::signal_data_t *data)
+    {
+        auto signal_data = (wf::output_added_signal*)data;
+        if (signal_data->output != output)
+        {
+            // This event is not for this output
+            return;
+        }
+
+        std::string identifier = make_output_identifier(output);
+
+        // Focus this output if it was the last one focused
+        if (core_get_focused_output() == identifier)
+        {
+            LOGD("This is last focused output, refocusing: ", output->to_string());
+            wf::get_core().focus_output(output);
+            core_erase_focused_output();
+        }
+
+        for (auto& wo : wf::get_core().output_layout->get_outputs())
+        {
+            // Make a list of views to move to this output
+            auto views = std::vector<wayfire_view>();
+            for (auto& view : wo->workspace->get_views_in_layer(wf::LAYER_WORKSPACE))
+            {
+                if ((view->role != wf::VIEW_ROLE_TOPLEVEL) || !view->is_mapped())
+                {
+                    continue;
+                }
+
+                if (!view_has_data(view))
+                {
+                    continue;
+                }
+
+                auto last_output_info = view_get_data(view);
+                if (last_output_info->output_identifier == identifier)
+                {
+                    views.push_back(view);
+                }
+            }
+
+            // Sorts with the views closest to front last
+            std::sort(views.begin(), views.end(), last_output_info_z_sort());
+
+            // Move views to this output
+            for (auto& view : views)
+            {
+                auto last_output_info = view_get_data(view);
+                LOGD("Restoring view: ",
+                    view->get_title(), " to: ", output->to_string());
+
+                wf::get_core().move_view_to_output(view, output, false);
+                view->set_fullscreen(last_output_info->fullscreen);
+                view->set_minimized(last_output_info->minimized);
+                view->set_sticky(last_output_info->sticky);
+                view->set_tiled(last_output_info->tiled_edges);
```

Comment:
hmm, so why do you need to first set the tiled edges and then emit the tile request? sounds counterintuitive to me.

Reply:
It wasn't tiling properly until I did this. I'll check if I need the set_tiled, I think I do because the tile_request is only emitted if tiled_edges != 0 - and if we emit tile_request when tiled_edges is 0, the grid plugin resizes the window and so stops our resize working.

Reply:
I've tested without set_tiled and everything still seems to work, so I've removed it.